crates/router/src/core/routing.rs

```diff
@@ -387,16 +386,16 @@ pub async fn retrieve_default_routing_config(
     metrics::ROUTING_RETRIEVE_DEFAULT_CONFIG.add(&metrics::CONTEXT, 1, &[]);
     let db = state.store.as_ref();
 
-    helpers::get_merchant_default_config(db, &merchant_account.merchant_id, transaction_type)
-        .await
-        .map(|conn_choice| {
-            metrics::ROUTING_RETRIEVE_DEFAULT_CONFIG_SUCCESS_RESPONSE.add(
-                &metrics::CONTEXT,
-                1,
-                &[],
-            );
-            service_api::ApplicationResponse::Json(conn_choice)
-        })
+    helpers::get_merchant_default_config(
+        db,
+        merchant_account.get_id().get_string_repr(),
+        transaction_type,
+    )
+    .await
+    .map(|conn_choice| {
+        metrics::ROUTING_RETRIEVE_DEFAULT_CONFIG_SUCCESS_RESPONSE.add(&metrics::CONTEXT, 1, &[]);
```

Comment:
I think it might be useful to have relevant attributes here, if merchant ID is one among them.